### DISCUSSION TITLE
does ps plan to implement a webhook functionality?

### DISCUSSION THREAD

**Discussion by bkyerv** [planetscale/discussion#361] [Feature Requests] [answered]
Is there any plans to implement a webhook functionality that can be triggerred by the db on certain events, e.g. new data is inserted/updated to the db? This is mostly for notification in case of changes to the tables so that the server can revalidate the cache?

**Comment by fooforge** [comment, marked as answer]
@bkyerv I would let the application handle this. It's the entity that controls and manages your data in the end, and as such it should know when to take certain actions. In our Rails app, for instance, we make heavy use of [Active Record Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) to trigger operations such as cache invalidations, search index updates, or audit logging. If you pass that responsibility to the database layer, chances are there will be inefficiencies.

**Comment by bkyerv** [comment]
I am a beginner software developer and was not even aware of it. Thank you. I guess I will need to do a thorough research on this subject

**Reply by fooforge** [reply]
You're welcome, @bkyerv.
